py/visdom/__init__.py

```diff
@@ -2128,6 +2128,52 @@ def stem(self, X, Y=None, win=None, env=None, opts=None):
 
         return self.scatter(X=data, Y=labels, opts=opts, win=win, env=env)
 
+
+    @pytorch_wrap
+    def sunburst(self, labels, parents, values=None, win=None, env=None, opts=None):
+        opts = {} if opts is None else opts
+        _title2str(opts)
+        _assert_opts(opts)
+
+        font_size=opts.get("size")
+        font_color=opts.get("font_color")
+        opacity=opts.get("opacity")
+        line_width=opts.get("marker_width")
+
+        assert len(parents.tolist())==len(labels.tolist()), "length of parents and labels should be equal"
+        
+        if(values!=None):
+            values = np.squeeze(values)
+            assert values.ndim == 1, 'values should be one-dimensional'
+            assert len(parents.tolist())==len(values.tolist()), "length of values should be equal to lenght of labels and parents"
+            data = [{
+                'values': values.tolist(),
+                'labels': labels.tolist(),
+                "parents":parents.tolist(),
+                "outsidetextfont":{"size":font_size,"color":font_color},
+                "leaf":{"opacity":opacity},
+                "marker":{"line":{"width":line_width}},
+                'type': 'sunburst',
+            }]
+        
+        else:
+            data = [{
+                'labels': labels.tolist(),
+                "parents":parents.tolist(),
+                "outsidetextfont":{"size":font_size,"color":font_color},
+                "leaf":{"opacity":opacity},
+                "marker":{"line":{"width":line_width}},
+                'type': 'sunburst',
+            }]            
```

Comment:
This is a particularly ineffective way to deal with this, as if you ever want to change the remains of the data packet you'd need to change in both locations.

You can do something like:
```
data_dict = {...}
if values is not None:
    ...
    data_dict['values'] = ...

data = [data_dict]
```